### PR TITLE
Fixes ESLint for Accounts

### PR DESCRIPTION
**Pull request by carson-ricca**
- Fixes all ESLint issues for the accounts component.

### CODE REVIEW

File: src/app/components/accounts/activation-email/activation-email.component.ts [hunk 1]
```diff
@@ -28,12 +29,12 @@ export class ActivationEmailComponent implements OnInit {
         });
 
         this.registerService.PostActivation(this.uuid, this.token)
-            .subscribe(response => {
+            .subscribe(() => {
                 this.messageService.add(MESSAGE_TYPES.SUCCESS, 'You have activated your account successfully.');
-                this.router.navigate(['/accounts/login']);
+                this.router.navigate(['/accounts/login']).then();
```
Comment (carson-ricca) [inline src/app/components/accounts/activation-email/activation-email.component.ts:34]: I'm not sure what we should put in the `.then()` or if we just leave it empty. It throws a linting warning so I feel like we should include it.

File: src/app/components/accounts/consent-form/consent-form.component.ts [hunk 2]
```diff
@@ -30,10 +31,10 @@ export class ConsentFormComponent implements OnInit {
         });
     }
 
-    onSubmit(FormData) {
+    onSubmit(FormData: unknown): void {
```
Comment (carson-ricca) [inline src/app/components/accounts/consent-form/consent-form.component.ts:34]: `unknown` is recommended by ESLint to use in place of `any`

File: src/app/components/accounts/activation-email/activation-email.component.ts (hunk 1 above)
Comment (keyvankhademi) [reply src/app/components/accounts/activation-email/activation-email.component.ts:34]: yeah `.then()` should be there.

File: src/app/components/accounts/consent-form/consent-form.component.ts (hunk 2 above)
Comment (keyvankhademi) [reply src/app/components/accounts/consent-form/consent-form.component.ts:34]: form data has a specific type inferred from the form.

File: src/app/components/accounts/profile-details/profile-details.component.ts [hunk 3]
```diff
@@ -41,24 +42,24 @@ export class ProfileDetailsComponent implements OnInit {
         });
     }
 
-    onSubmit(FormData) {
+    onSubmit(FormData: unknown): void {
```
Comment (keyvankhademi) [inline src/app/components/accounts/profile-details/profile-details.component.ts:45]: same here

File: src/app/components/accounts/register/register.component.ts [hunk 4]
```diff
@@ -1,23 +1,22 @@
...
 import {MESSAGE_TYPES} from '@app/_models';
-import {AuthenticationService} from '@app/_services/api/authentication';
-import {first} from 'rxjs/operators';
 
 @Component({
     selector: 'app-register',
     templateUrl: './register.component.html',
-    styleUrls: ['./register.component.scss']
+    styleUrls: ['./register.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
```
Comment (keyvankhademi) [inline src/app/components/accounts/register/register.component.ts:13]: what is this?

Comment (carson-ricca) [reply src/app/components/accounts/register/register.component.ts:13]: It was required by the ESLint checker. It throws and error without this. This is what was recommended to include.

File: src/app/components/accounts/profile-details/profile-details.component.ts (hunk 3 above)
Comment (carson-ricca) [reply src/app/components/accounts/profile-details/profile-details.component.ts:45]: Do you know the type? I tried doing a map and it said that unknown should be used instead.

File: src/app/components/accounts/consent-form/consent-form.component.ts (hunk 2 above)
Comment (keyvankhademi) [reply src/app/components/accounts/consent-form/consent-form.component.ts:34]: the type of formData should be `FormGroup` and use `.get` to access its members.

File: src/app/components/accounts/register/register.component.ts (hunk 4 above)
Comment (keyvankhademi) [reply src/app/components/accounts/register/register.component.ts:13]: I removed this from eslint. You can change it back.

File: src/app/components/accounts/consent-form/consent-form.component.ts (hunk 2 above)
Comment (carson-ricca) [reply src/app/components/accounts/consent-form/consent-form.component.ts:34]: Okay I see that makes more sense.

File: src/app/components/accounts/register/register.component.ts (hunk 4 above)
Comment (carson-ricca) [reply src/app/components/accounts/register/register.component.ts:13]: Okay cool

**Review by keyvankhademi — APPROVED** [review]

**Pull request merged**